server/game/costs.js

```diff
@@ -597,6 +598,37 @@ const Costs = {
                 context.game.addGold(context.player, -amount);
             }
         };
+    },
+    /**
+     * Cost where the player gets prompted to pay from 1 up to the lesser of two values: 
+     * the passed value and either the player's or his opponent's gold.
+     * Used by Ritual of R'hllor and Loot.
+     * TODO: needs to be reducable for cards like Littlefinger's Meddling and Paxter Redwyne.
+     */
+    payXGold: function(maxFunc, opponentObj = false) {
+        return {
+            canPay: function(context) {
+                if(!opponentObj) {
+                    return context.player.gold >= 1;
+                }
+                return opponentObj.gold >= 1;
+            },
+            resolve: function(context, result = { resolved: false }) {
+                let gold = opponentObj ? opponentObj.gold : context.player.gold;
+                let list = [maxFunc(), gold];
```

Comment:
This is non-intuitive. Why not calculate the limit here (you have enough info) and pass it into the prompt directly? The array doesn't seem to be used for anything else.